### PR TITLE
Feat/add toned theme

### PR DESCRIPTION
**Pull request by Kreezag**
## Description
part of [vuestic-admin pr #662](https://github.com/epicmaxco/vuestic-admin/issues/662)

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

Changes in app:
 - Implement toned down theme.
- Theme should have no gradients.
- Theme should have less shadows.
- Update color-theme-plugin for custom color sets

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,19037 @@
+{
```
Comment (asvae) [inline package-lock.json]: Please remove

File: src/components/vuestic-components/va-color-presentation/VaColorPresentation.vue [hunk 2]
```diff
@@ -63,41 +63,34 @@ export default {
...
   },
   methods: {
     colorCopy () {
-      if (this.variant.includes('gradient')) {
-        this.$copyText(getGradientBackground(this.colorComputed))
-        return
-      }
-
-      this.$copyText(this.colorComputed)
+      this.$copyText(this.calcBackground)
```
Comment (asvae) [inline src/components/vuestic-components/va-color-presentation/VaColorPresentation.vue]: Pretty sure this won't work. Please test.

File: src/components/vuestic-components/va-data-table/VaDataTable.vue [hunk 3]
```diff
@@ -107,6 +106,9 @@ export default {
         },
       }
     },
+    paginationStyles () {
```
Comment (asvae) [inline src/components/vuestic-components/va-data-table/VaDataTable.vue]: just `color`? There are no styles.

File: src/components/vuestic-components/va-data-table/VaDataTable.vue [hunk 4]
```diff
@@ -107,6 +106,9 @@ export default {
...
+      return this.isDefaultColorTheme ? 'info' : 'primary'
```
Comment (asvae) [inline src/components/vuestic-components/va-data-table/VaDataTable.vue]: This doesn't seem right to me. Theme color should stay consistent.

File: src/components/vuestic-components/va-select/VaSelect.vue [hunk 5]
```diff
@@ -258,6 +258,11 @@ export default {
         paddingBottom: this.label ? 0 : this.multiple ? '.3125rem' : '.4375rem',
       }
     },
+    labelStyles () {
```
Comment (asvae) [inline src/components/vuestic-components/va-select/VaSelect.vue]: `labelStyle`? Why styles?

File: src/components/vuestic-components/va-sidebar/VaSidebarLink.vue [hunk 6]
```diff
@@ -87,16 +87,25 @@ export default {
 
       if (this.isHovered || this.isActive) {
         return {
-          color: this.$themes['primary'],
-          backgroundColor: getBackgroundColor(),
-          borderColor: this.isActive ? this.$themes['primary'] : 'transparent',
+          color: this.isDefaultColorTheme ? this.$themes.primary : 'white',
```
Comment (asvae) [inline src/components/vuestic-components/va-sidebar/VaSidebarLink.vue]: That is not ideal way to handle this hack.

More correct way would be to do a computed, like `noGradient`, `invertFontColor`, `shadowType`. As that would be consistent with what we plan to achieve with context.

Please don't fix. Just for your information.

File: src/components/vuestic-components/va-sidebar/VaSidebarLinkGroup.vue [hunk 7]
```diff
@@ -11,7 +11,7 @@
         <va-icon
           v-if="icon"
           class="va-sidebar-link__content__icon"
-          :style="iconStyles"
+          :style="computedIconStyles"
```
Comment (asvae) [inline src/components/vuestic-components/va-sidebar/VaSidebarLinkGroup.vue]: style

File: src/components/vuestic-components/va-sidebar/VaSidebarLinkGroup.vue [hunk 8]
```diff
@@ -131,7 +131,7 @@ export default {
       this.isHovered = !this.isHovered
     },
     updateActiveState () {
-      const active = this.children.some(item => item.name === this.$route.name)
+      const active = this.children && this.children.some(item => item.name === this.$route.name)
```
Comment (asvae) [inline src/components/vuestic-components/va-sidebar/VaSidebarLinkGroup.vue]: What's the point in this check?

File: src/components/vuestic-components/va-sidebar/VaSidebarLinkGroup.vue [hunk 9]
```diff
@@ -166,16 +166,26 @@ export default {
...
+
+      return {
+        color: this.isDefaultColorTheme ? 'rgba(255, 255, 255, 0.65)' : this.$themes.secondary,
+      }
     },
-    iconStyles () {
-      return (this.isHovered || this.isActive)
-        ? { color: this.$themes['primary'] }
-        : { color: 'white' }
+    computedIconStyles () {
```
Comment (asvae) [inline src/components/vuestic-components/va-sidebar/VaSidebarLinkGroup.vue]: style

File: src/components/vuestic-plugin.js [hunk 10]
```diff
@@ -11,6 +11,7 @@ import VaChart from './vuestic-components/va-chart/VaChart.vue'
 import VaCheckbox from './vuestic-components/va-checkbox/VaCheckbox.vue'
 import VaChip from './vuestic-components/va-chip/VaChip.vue'
 import VaCollapse from './vuestic-components/va-collapse/VaCollapse'
+import VaColorPresentation from './vuestic-components/va-color-presentation/VaColorPresentation'
```
Comment (asvae) [inline src/components/vuestic-plugin.js]: This component is not needed in vuestic-ui.

File: src/services/ColorThemePlugin.js [hunk 11]
```diff
@@ -54,5 +79,22 @@ export const ColorThemeMixin = {
       }
       return this.colorDefault
     },
+    isDefaultColorTheme () {
+      return this.$themesOptions.activeThemeName === this.$themesOptions.defaultThemeName
+    },
+  },
+}
+
+export const ColorThemeActionsMixin = {
```
Comment (asvae) [inline src/services/ColorThemePlugin.js]: This code is never used.

File: src/services/ColorThemePlugin.js [hunk 12]
```diff
@@ -12,18 +12,43 @@ const getDefaultOptions = () => ({
...
+      activeThemeName: defaultThemeName,
+      defaultThemeName: defaultThemeName,
+      themesList: optionsThemesByName,
+    }
 
     /* eslint-disable no-new */
-    // This line is just to make themes reactive
-    new Vue({ data: { themes: Vue.prototype.$themes } })
+    // This line is just to make fields reactive
+    new Vue({ data: {
```
Comment (asvae) [inline src/services/ColorThemePlugin.js]: Use Vue.observable

File: src/services/ColorThemePlugin.js [hunk 13]
```diff
@@ -12,18 +12,43 @@ const getDefaultOptions = () => ({
 })
 
 export const ColorThemePlugin = {
-  install (Vue, options) {
+  install (Vue, options) { // options = [ { name: 'myName', themes: { primary: '', ... }} ]
```
Comment (asvae) [inline src/services/ColorThemePlugin.js]: Multiple themes are never needed. Should be one object.

File: src/services/ColorThemePlugin.js [hunk 14]
```diff
@@ -12,18 +12,43 @@ const getDefaultOptions = () => ({
...
+      options = [{ name: 'default', ...defaultOptions }]
     }
 
-    Vue.prototype.$themes = defaultOptions.themes
+    if (options && !Array.isArray(options)) {
+      throw new Error('Options of ColorThemePlugin should be an array')
+    }
+
+    if (!options[0].name || !options[0].themes) {
+      throw new Error('Options should has name and themes props')
```
Comment (asvae) [inline src/services/ColorThemePlugin.js]: *should have

File: src/services/ColorThemePlugin.js (hunk 11 above)
Comment (Kreezag) [reply src/services/ColorThemePlugin.js]: This code used in vuestic-admin components. Added comment

File: src/services/ColorThemePlugin.js (hunk 13 above)
Comment (Kreezag) [reply src/services/ColorThemePlugin.js]: In vuestic-admin used multiple themes (original/corporate)
These changes need for possibility change the corporate theme by user

File: src/components/vuestic-components/va-sidebar/VaSidebarLinkGroup.vue (hunk 8 above)
Comment (Kreezag) [reply src/components/vuestic-components/va-sidebar/VaSidebarLinkGroup.vue]: In demo children equal undefined

File: src/components/vuestic-components/va-sidebar/VaSidebarLink.vue (hunk 6 above)
Comment (Kreezag) [reply src/components/vuestic-components/va-sidebar/VaSidebarLink.vue]: This component not used in vuestic-admin. May be need to remove va-sidebar components from vuestic-ui

File: src/components/vuestic-components/va-data-table/VaDataTable.vue (hunk 4 above)
Comment (Kreezag) [reply src/components/vuestic-components/va-data-table/VaDataTable.vue]: These changes proposed by @xx13 
`в альтернативной теме primary цвет используется как фон в навигации. (c)`

Comment (Kreezag) [reply src/components/vuestic-components/va-data-table/VaDataTable.vue]: These changes proposed by @xx13 
`в альтернативной теме primary цвет используется как фон в навигации. (c)`

Comment (Kreezag) [reply src/components/vuestic-components/va-data-table/VaDataTable.vue]: These changes proposed by @xx13 
`в альтернативной теме primary цвет используется как фон в навигации. (c)`

**Pull request merged**